src/SNPN/Common/NetworkService.cs

```diff
@@ -19,21 +23,42 @@ public class NetworkService : INetworkService, IDisposable
 		private readonly AppConfiguration _config;
 		private readonly HttpClient _httpClient;
 		private readonly ILogger _logger;
+		private readonly IUserRepo _userRepo;
 
 		private readonly AsyncPolicy _retryPolicy;
 
+		private static Object firebaseAppLock = new Object();
+
 		private readonly Dictionary<NotificationType, string> _notificationTypeMapping = new Dictionary<NotificationType, string>
 		  {
 				{ NotificationType.Badge, "wns/badge" },
 				{ NotificationType.Tile, "wns/tile" },
 				{ NotificationType.Toast, "wns/toast" }
 		  };
 
-		public NetworkService(AppConfiguration configuration, ILogger logger, HttpClient httpClient)
+		public NetworkService(AppConfiguration configuration, ILogger logger, HttpClient httpClient, IUserRepo userRepo)
 		{
 			_config = configuration;
 			_logger = logger;
 			_httpClient = httpClient;
+			_userRepo = userRepo;
+
+			// Setup Firebase Default App with credentials only once.
+			lock(firebaseAppLock) {
+				var fcmJSON = Environment.GetEnvironmentVariable("FCM_KEY_JSON");
+				if(fcmJSON != null) {
+					if (FirebaseApp.DefaultInstance == null) {
+						FirebaseApp.Create(new AppOptions()
```

Comment:
Setup the default Firebase app with the credentials passed in through the JSON. `FCM_KEY_JSON` is the environment variable that the content is retrieved from.